src/raftstore/store/snap.rs

```diff
@@ -1001,20 +1021,44 @@ impl Write for Snap {
 
             let file = cf_file.file.as_mut().unwrap();
```

Comment:
You can use the `LimitWriter` here, like

```
let file = match self.limiter {
    Some(limiter) => LimitWriter::new(cf_file.file),
    None => cf_file.file,
};
```

Reply:
```
let file = match self.limiter {
    Some(limiter) => LimitWriter::new(limiter, cf_file.file.as_mut().unwrap()),
    None => cf_file.file.as_mut().unwrap(),
};
```

I can't do it like this because they are two different types. Instead I put `Option<IOLimiter>` in `LimitWriter` so that we can choose whether to limite io in `LimitWriter`.